src/reactionsystem_structure.jl

```diff
@@ -0,0 +1,1438 @@
+        sts = unique!([sts; reduce(vcat, namespace_variables.(systems))])
+        sort!(sts; by = !isspecies)
+        return sts
+    end
+    return sts
+end
+
+
+### Basic `ReactionSystem`-specific Accessors ###
+
+"""
+    get_species(sys::ReactionSystem)
+
+Return the current dependent variables that represent species in `sys` (toplevel system
+only).
+"""
+get_species(sys::ReactionSystem) = getfield(sys, :species)
+has_species(sys::ReactionSystem) = isdefined(sys, :species)
+
+"""
+    get_rxs(sys::ReactionSystem)
+
+Return the system's `Reaction` vector (toplevel system only).
+"""
+get_rxs(sys::ReactionSystem) = getfield(sys, :rxs)
+has_rxs(sys::ReactionSystem) = isdefined(sys, :rxs)
+
+"""
+    get_sivs(sys::ReactionSystem)
+
+Return the current spatial ivs, if the system is non-spatial returns an empty vector.
+"""
+get_sivs(sys::ReactionSystem) = getfield(sys, :sivs)
+has_sivs(sys::ReactionSystem) = isdefined(sys, :sivs)
+
+"""
+    get_networkproperties(sys::ReactionSystem)
+
+Return the current network properties of `sys`.
+"""
+get_networkproperties(sys::ReactionSystem) = getfield(sys, :networkproperties)
+
+"""
+    get_combinatoric_ratelaws(sys::ReactionSystem)
+
+Returns true if the default for the system is to rescale ratelaws, see
+https://docs.sciml.ai/Catalyst/stable/introduction_to_catalyst/introduction_to_catalyst/#Reaction-rate-laws-used-in-simulations
+for details. Can be overriden via passing `combinatoric_ratelaws` to `convert` or the
+`*Problem` functions.
+"""
+get_combinatoric_ratelaws(sys::ReactionSystem) = getfield(sys, :combinatoric_ratelaws)
+
+"""
+    combinatoric_ratelaws(sys::ReactionSystem)
+
+Returns the effective (default) `combinatoric_ratelaw` value for a compositional system,
+calculated by taking the logical or of each component `ReactionSystem`. Can be overriden
+during calls to `convert` of problem constructors.
+"""
+function combinatoric_ratelaws(sys::ReactionSystem)
+    crl = get_combinatoric_ratelaws(sys)
+    subsys = Iterators.filter(s -> s isa ReactionSystem, get_systems(sys))
+    mapreduce(combinatoric_ratelaws, |, subsys; init = crl)
+end
+
+"""
+    species(network)
+
+Given a [`ReactionSystem`](@ref), return a vector of all species defined in the system and
+any subsystems that are of type `ReactionSystem`. To get the species and non-species
+variables in the system and all subsystems, including non-`ReactionSystem` subsystems, uses
+`unknowns(network)`.
+
+Notes:
+- If `ModelingToolkit.get_systems(network)` is non-empty will allocate.
+"""
+function species(network)
+    sts = get_species(network)
+    systems = filter_nonrxsys(network)
+    isempty(systems) && return sts
+    unique([sts; reduce(vcat, map(sys -> species(sys, species(sys)), systems))])
+end
+
+# Special species but which take a set of states and names spaces them according to another
+# `ReactionSystem`.
+function species(network::ReactionSystem, sts)
+    [MT.renamespace(network, st) for st in sts]
+end
```

Comment:
order...

Reply:
Figured they were both species functions, so I had this order, but can move if you want.

Reply:
Order of use. This version is called by the other version so should precede it.